engine/table/src/main/java/io/deephaven/engine/util/PyCallableWrapper.java

```diff
@@ -0,0 +1,266 @@
+package io.deephaven.engine.util;
+
+import io.deephaven.engine.table.impl.select.python.ArgumentsChunked;
+import org.jpy.PyModule;
+import org.jpy.PyObject;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * When given a pyObject that is a callable, we stick it inside the callable wrapper, which implements a call() varargs
+ * method, so that we can call it using __call__ without all of the JPy nastiness.
+ */
+public class PyCallableWrapper {
+    private static final PyObject NUMBA_VECTORIZED_FUNC_TYPE = PyCallableWrapper.getNumbaVectorizedFuncType();
+
+    private static final PyModule dh_table_module = PyModule.importModule("deephaven.table");
+
+    private static final Map<Character, Class<?>> numpyType2JavaClass = new HashMap<>();
+
+    static {
+        numpyType2JavaClass.put('i', int.class);
+        numpyType2JavaClass.put('l', long.class);
+        numpyType2JavaClass.put('h', short.class);
+        numpyType2JavaClass.put('f', float.class);
+        numpyType2JavaClass.put('d', double.class);
+        numpyType2JavaClass.put('b', byte.class);
+        numpyType2JavaClass.put('?', boolean.class);
+        numpyType2JavaClass.put('O', Object.class);
+    }
+
+    private PyObject pyCallable;
+
+    private String signature = null;
+    private List<Class<?>> paramTypes;
+    private Class<?> returnType;
+    private boolean vectorizable = false;
+    private boolean vectorized = false;
+
+    // first value of the pair is null if it is a resolved constant arg, otherwise it is the used column name when
+    // uninitialized and an index value of the chunked arrays for the column sources used as args when initialized
+    private Collection<ChunkArgument> chunkArguments;
+    private Class<?>[] chunkArgumentTypes;
+
+    private boolean numbaVectorized;
+    private PyObject unwrapped;
+
+    public PyCallableWrapper(PyObject pyCallable) {
+        this.pyCallable = pyCallable;
+    }
+
+    public ArgumentsChunked buildArgumentsChunked(List<String> columnSources) {
+        Object[] args;
+        Class<?>[] argTypes;
+
+        if (!numbaVectorized) {
+            args = new Object[chunkArguments.size() + 1];
+            argTypes = new Class[chunkArgumentTypes.length + 1];
+            argTypes[0] = int.class;
+        } else {
+            args = new Object[chunkArguments.size()];
+            argTypes = new Class[chunkArgumentTypes.length];
+        }
```

Comment:
Here is more to the `ArgumentsChunked` puzzle.  It looks like the logic is split up in `ArgumentsChunked` and here, which is probably not ideal.  This logic applies to `args`, `argTypes` allocations as well as the numba flag. This is also where `argTypes[0]` is set.

Reply:
Should this method logic be in `ArgumentsChunked`?